lib/ConnectionContextBase.ts

```diff
@@ -7,6 +7,8 @@ import { DataTransformer, DefaultDataTransformer } from "./dataTransformer";
 import { TokenProvider } from "./auth/token";
 import { ConnectionConfig } from "./connectionConfig";
 import { SasTokenProvider } from "./auth/sas";
+import rhea from "rhea";
```

Comment:
@amarzavery had some concerns using `rhea` directly in any of our code instead of using `rhea-promise`. I am not sure on what they were, I just recall that they were present.

If the `rhea` version resolved in `amqp-common` doesnt match with that in `rhea-promise`, we might end up with 2 versions of `rhea` correct?

Wasnt this was the reason why `rhea-promise` is a peer dependency in `amqp-common`?
If so, then `rhea` should be a peer dependency as well.